engine/components-api/src/com/cloud/agent/AgentManager.java

```diff
@@ -141,7 +142,7 @@
 
     public void pullAgentOutMaintenance(long hostId);
 
-    boolean reconnect(long hostId);
+    void reconnect(long hostId) throws CloudRuntimeException, AgentUnavailableException;
```

Comment:
Are you talking in a philosophical/ideological way? Because `CloudRuntimeException` is a runtime exception, so the compiler will not complain if the code calls the method and does not handle the declared exception.